In [4]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns 

## 1. Data Loading 

In [5]:

files = ["2011.csv","2012.csv","2013.csv","2014.csv","2015.csv"]
dfs = {year : pd.read_csv(f"C:/Users/Vansh/Desktop/Project/data/raw/{f}") for year,f in zip(range(2011,2016),files)}

for year,df in dfs.items():
    print(f"{year}:{df.shape} - {list(df.columns)}")


2011:(506467, 454) - ['_STATE', '_GEOSTR', '_DENSTR2', 'PRECALL', 'REPNUM', 'REPDEPTH', 'FMONTH', 'IDATE', 'IMONTH', 'IDAY', 'IYEAR', 'INTVID', 'DISPCODE', 'SEQNO', '_PSU', 'NATTMPTS', 'NRECSEL', 'NRECSTR', 'CTELENUM', 'CELLFON', 'PVTRESID', 'NUMADULT', 'NUMMEN', 'NUMWOMEN', 'GENHLTH', 'PHYSHLTH', 'MENTHLTH', 'POORHLTH', 'HLTHPLN1', 'PERSDOC2', 'MEDCOST', 'CHECKUP1', 'BPHIGH4', 'BPMEDS', 'BLOODCHO', 'CHOLCHK', 'TOLDHI2', 'CVDINFR4', 'CVDCRHD4', 'CVDSTRK3', 'ASTHMA3', 'ASTHNOW', 'CHCSCNCR', 'CHCOCNCR', 'CHCCOPD', 'HAVARTH3', 'ADDEPEV2', 'CHCKIDNY', 'CHCVISON', 'DIABETE3', 'SMOKE100', 'SMOKDAY2', 'STOPSMK2', 'LASTSMK2', 'USENOW3', 'AGE', 'HISPANC2', 'MRACE', 'ORACE2', 'VETERAN3', 'MARITAL', 'CHILDREN', 'EDUCA', 'EMPLOY', 'INCOME2', 'WEIGHT2', 'HEIGHT3', 'CTYCODE1', 'NUMHHOL2', 'NUMPHON2', 'CPDEMO1', 'CPDEMO2', 'CPDEMO3', 'CPDEMO4', 'RENTHOM1', 'SEX', 'PREGNANT', 'FRUITJU1', 'FRUIT1', 'FVBEANS', 'FVGREEN', 'FVORANG', 'VEGETAB1', 'EXERANY2', 'EXRACT01', 'EXEROFT1', 'EXERHMM1', 'EXRACT02', 

In [6]:
common_cols = set(dfs[2011].columns)
for df in dfs.values():
    common_cols = common_cols.intersection(set(df.columns))

print(f"Common columns: {len(common_cols)}")
print(common_cols)

Common columns: 157
{'PNEUVAC3', '_BMI5', 'NUMHHOL2', '_EDUCAG', 'HAVARTH3', 'PAINACT2', 'RENTHOM1', 'PHYSHLTH', 'WEIGHT2', 'SCNTLPAD', 'PREDIAB1', 'CTELENUM', 'DIABEDU', 'MENTHLTH', 'CASTHNO2', '_RFSEAT2', 'SCNTWRK1', 'MSCODE', 'EYEEXAM', 'RCSRLTN2', 'AVEDRNK2', 'CVDSTRK3', '_ASTHMS1', '_TOTINDA', 'USENOW3', 'MARITAL', '_LLCPWT', '_RAWRAKE', 'LASTPAP2', 'WTKG3', 'QSTLANG', 'EXERANY2', 'DROCDY3_', 'PVTRESD2', 'HPVADSHT', 'PROFEXAM', '_AGEG5YR', 'QSTVER', 'CSTATE', '_INCOMG', 'EMTSUPRT', 'CPDEMO1', 'BLDSUGAR', '_RFBMI5', 'LENGEXAM', '_HCVU651', 'ASATTACK', 'NUMWOMEN', 'ASRCHKUP', 'HADHYST2', '_PNEUMO2', 'VETERAN3', 'POORHLTH', 'ASTHNOW', 'NUMADULT', 'CADULT', 'LASTSMK2', '_LTASTH1', 'NUMPHON2', 'HIVTST6', '_RFHLTH', 'CHECKUP1', 'MAXDRNKS', 'DRNK3GE5', 'LSATISFY', 'RCSGENDR', 'ASTHMED3', 'INCOME2', 'CELLFON2', 'SEQNO', 'ALCDAY5', 'QLACTLM2', '_CLLCPWT', '_RFSEAT3', '_AIDTST3', 'LANDLINE', '_SMOKER3', 'INSULIN', 'CVDINFR4', 'LSTBLDS3', '_PSU', 'SCNTPAID', '_RFBING5', 'DRNKANY5', 'IDAY', '

In [7]:
for year,df in dfs.items():
    df["Survey_year"] = year

## 2. Data Merging 

In [8]:
merged_df = pd.concat([df[list(common_cols)+["Survey_year"]] for df in dfs.values()],ignore_index=True)

merged_df.to_csv("C:/Users/Vansh/Desktop/Project/data/raw/merged.csv",index=False)

In [9]:
print(merged_df.shape)
print(merged_df.head())

(2380047, 158)
   PNEUVAC3   _BMI5  NUMHHOL2  _EDUCAG  HAVARTH3  PAINACT2  RENTHOM1  \
0       7.0  1855.0       2.0      2.0       1.0       NaN       1.0   
1       NaN  3529.0       2.0      1.0       1.0       NaN       1.0   
2       2.0  1695.0       2.0      4.0       2.0       NaN       1.0   
3       NaN  2579.0       2.0      2.0       2.0       NaN       1.0   
4       2.0  4346.0       2.0      1.0       1.0       NaN       2.0   

   PHYSHLTH  WEIGHT2  SCNTLPAD  ...  _DRDXAR1  USEEQUIP  CVDCRHD4  CHCSCNCR  \
0      88.0     95.0       NaN  ...       1.0       2.0       2.0       2.0   
1      12.0    239.0       NaN  ...       1.0       NaN       2.0       2.0   
2      88.0    105.0       NaN  ...       2.0       2.0       2.0       1.0   
3      88.0    155.0       NaN  ...       2.0       NaN       2.0       2.0   
4      25.0    230.0       NaN  ...       1.0       2.0       2.0       2.0   

   CHKHEMO3  SMOKE100  ASNOSLEP  CHILDREN  HEIGHT3  Survey_year  
0       NaN

In [ ]:
if "DIABETE3" in list(merged_df.columns):
    print(True)

print(merged_df['DIABETE3'].value_counts())

merged_df["DIABETE3"] = merged_df["DIABETE3"].replace({7.0:np.nan,9.0:np.nan})
merged_df = merged_df.dropna(subset=["DIABETE3"])
print(merged_df.shape)
print(merged_df["DIABETE3"].value_counts())


True
DIABETE3
3.0    2010686
1.0     302943
4.0      39868
2.0      21317
Name: count, dtype: int64
(2374814, 159)
DIABETE3
3.0    2010686
1.0     302943
4.0      39868
2.0      21317
Name: count, dtype: int64


## 3. Target Variable Analysis

In [20]:
mapping = {1.0:1,2.0:1,4.0:1,3.0:0}
merged_df["Diabetes_binary"] = merged_df["DIABETE3"].map(mapping)
print(merged_df[["DIABETE3","Diabetes_binary"]])
print(merged_df["Diabetes_binary"].value_counts())

         DIABETE3  Diabetes_binary
0             3.0                0
1             3.0                0
2             3.0                0
3             3.0                0
4             4.0                1
...           ...              ...
2380042       1.0                1
2380043       3.0                0
2380044       3.0                0
2380045       3.0                0
2380046       1.0                1

[2374814 rows x 2 columns]
Diabetes_binary
0    2010686
1     364128
Name: count, dtype: int64


### Decision Documenting for Target Variable
1) The original values in the "DIABETE3" column were 1,2,3,4,7,9 where 
    - 1: Diabetic Patients
    - 2: Diabetic (While Preganant)
    - 3: Not Diabetic
    - 4: Pre Diabetic 
    - 7,9 were sentinel values showing "Refused" or "Not Answered"
2) We dropped 7 and 9 as they were a classic example of "Coded by the data Collector". Dropping these rows results in a loss of only 5,233 rows (~0.17% of data) — negligible at this scale. These values werent significant for our model training as they are irrelevant for classification
3) We Mapped 4 → 1 as this system is a risk prediction tool, not a diagnosis tool.Flagging pre-diabetics as high risk aligns with my stated goal of enabling lifestyle intervention before disease onset.
4) The final Class Distribution is
    - Class 0: 2010686 (84%)
    - Class 1: 364128 (16%)
5) This is a high class imbalance. I plan to handle this imbalance using mainly SMOTE for fixing the data , using "class_weight='balanced'" for fixing the model and lowering the decision threshold (flagging more people as positive) for fixing the threshold